File: 3731-find-missing-elements/3731-find-missing-elements.java
class Solution {
    public List<Integer> findMissingElements(int[] nums) {
        Arrays.sort(nums);


        Set<Integer> st = new HashSet<>();
        List<Integer> li = new ArrayList<>();

        for(int val : nums){
            st.add(val);
        }

        int str = nums[0];
        int end = nums[nums.length - 1];

        for(int i = str; i < end; i++){
            if(!st.contains(i)) li.add(i);
        }
        return li;
    }
}

File: 3731-find-missing-elements/3731-find-missing-elements.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var findMissingElements = function(nums) {
    const n = nums.length;

    const st = new Set();

    for(let i = 0; i < n; i++){
        st.add(nums[i])
    }

    nums.sort((a,b) => a - b);

    const arr = new Array();

    for(let i = nums[0]; i < nums[n - 1]; i++){

        if(!st.has(i)) arr.push(i);
    }

    return arr;
};